substrate/frame/parameters/src/tests/unit.rs

```diff
@@ -0,0 +1,237 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Unit tests for the parameters pallet.
+
+#![cfg(test)]
+
+use crate::tests::mock::{
+	dynamic_params::*, new_test_ext, PalletParameters, Runtime, RuntimeOrigin as Origin,
+	RuntimeParameters, RuntimeParameters::*, RuntimeParametersKey, RuntimeParametersValue,
+};
+use codec::Encode;
+use frame_support::{
+	assert_noop, assert_ok, traits::dynamic_params::AggregratedKeyValue, StorageNoopGuard,
+};
+use sp_core::Get;
+use sp_runtime::DispatchError;
+
+#[docify::export]
+#[test]
+fn set_parameters_example() {
+	new_test_ext().execute_with(|| {
+		assert_eq!(pallet1::Key3::get(), 2, "Default works");
+
+		// This gets rejected since the origin is not root.
+		assert_noop!(
+			PalletParameters::set_parameter(
+				Origin::signed(1),
+				Pallet1(pallet1::Parameters::Key3(pallet1::Key3, Some(123))),
+			),
+			DispatchError::BadOrigin
+		);
+
+		assert_ok!(PalletParameters::set_parameter(
+			Origin::root(),
+			Pallet1(pallet1::Parameters::Key3(pallet1::Key3, Some(123))),
+		));
+
+		assert_eq!(pallet1::Key3::get(), 123, "Update works");
```

Comment:
Maybe check that the `Event::<T>::Updated` has been thrown here too as expected?